src/wp-includes/formatting.php

```diff
@@ -2132,19 +2132,17 @@ function sanitize_user( $username, $strict = false ) {
  *
  * @since 3.0.0
  *
- * @param string $key String key
- * @return string Sanitized key
+ * @param string $key String key.
+ * @return string Sanitized key.
  */
 function sanitize_key( $key ) {
 	$raw_key = $key;
 
-	if ( ! is_string( $key ) ) {
-		$key = '';
-	}
-
-	if ( '' !== $key ) {
+	if ( is_scalar( $key ) ) {
 		$key = strtolower( $key );
 		$key = preg_replace( '/[^a-z0-9_\-]/', '', $key );
+	} else {
+		$key = '';
 	}
```

Comment:
Refactor query. Would it be cleaner/clearer to do something like this?

```php
function sanitize_key( $key ) {
    $sanitized_key = '';

    if ( is_scalar( $key ) ) {
        $sanitized_key = strtolower( $key );
        $sanitized_key = preg_replace( '/[^a-z0-9_\-]/', '', $key );
    }

    return apply_filters( 'sanitize_key', $sanitized_key, $key );
}
```

Reply:
Nice. Committed here https://github.com/WordPress/wordpress-develop/pull/2042/commits/245d6d3987ef7506d6956fbdb2eb1e1938848bcb.